embed/etcd.go

```diff
@@ -384,13 +384,18 @@ func (e *Etcd) serve() (err error) {
 		}(pl)
 	}
 
+	var metricsMux *http.ServeMux
+	if e.Config().ExposeMetrics {
+		metricsMux = http.NewServeMux()
+	}
+
 	// Start a client server goroutine for each listen address
 	var h http.Handler
 	if e.Config().EnableV2 {
-		h = v2http.NewClientHandler(e.Server, e.Server.Cfg.ReqTimeout())
+		h = v2http.NewClientHandler(e.Server, metricsMux, e.Server.Cfg.ReqTimeout())
```

Comment:
`NewClientHandler` and `HandleBasic` shouldn't need to change. I think all that's needed is:
```go
var metricsMux *http.ServeMux
if len(e.cfg.MetricsUrls) > 0 {
    metricsMux = http.NewServeMux()
    metricsMux.Handle("metrics", prometheus.Handler()) // maybe etcdhttp.MetricsPath or get the path from the user-provided URL
}
```